src/Serializing/SExpression/Precise.hs

```diff
@@ -78,5 +78,5 @@ instance ToSExpression t => GToSExpression (Rec1 t) where
 
 instance (Foldable f, GToSExpression g) => GToSExpression (f :.: g) where
   gtoSExpression (Comp1 fs) n
-    | null fs   = [nl n <> pad n <> "[]"]
-    | otherwise = nl n <> pad n <> "[" : foldMap gtoSExpression fs (n + 1) <> ["]"]
+    | null fs   = mempty
+    | otherwise = foldMap gtoSExpression fs n
```

Comment:
Is the intention here to mimic the existing s-expression output’s style for tests? I’d actually prefer if we _improved_ on that style, in general, since it doesn’t e.g. make it clear when there are multiple subterms in a single field, or when there is a field with no subterms. (I seem to recall we had an issue about this at one point, but I can’t find it just now.) Thoughts?